internal/venice-common/src/main/java/com/linkedin/venice/helix/HelixLiveInstanceMonitor.java

```diff
@@ -49,18 +49,23 @@ private String getLiveInstanceRootPath() {
     return HelixUtils.getHelixClusterZkPath(clusterName) + LIVE_INSTANCE_PATH;
   }
 
+  /**
+   * Must be run during service start else it wont subscribe to state changes.
+   */
   @Override
   public void refresh() {
     // subscribe is thread safe method.
-    zkClient.subscribeStateChanges(zkStateListener);
-    zkClient.subscribeChildChanges(getLiveInstanceRootPath(), this);
     /**
      * Here is using {@link ZkClient#getChildren(String)}, which should either success or fail.
      * It should not be possible to receive partial data here.
      */
-    List<String> instances = zkClient.getChildren(getLiveInstanceRootPath());
-    liveInstanceSet = convertToInstance(instances);
-    LOGGER.info("Got live instances: " + liveInstanceSet);
+    if (zkClient.exists(getLiveInstanceRootPath())) {
```

Comment:
Shall we fail fast if the required zk path doesn't exist?
If this class doesn't subscribe this zk path and this path is created later, it won't receive any updates about live instances, right?

Reply:
I added this for quite a few multi-cluser tests which fails since the path was not ready. So for prod it would be fail-fast, but for test, we can add some tries ?

Reply:
Can you check when this zk path gets created? During cluster initialization time or the first storage node initialization?

Reply:
It wasnt ready during cluster initialization. Must be after that.